tests/onnx/test_passes.py

```diff
@@ -21,3 +27,43 @@ def test_apply_preprocess_passes():
 
     assert set(after_nodes) - set(before_nodes) == set()
     assert set(before_nodes) - set(after_nodes) == set(["cast"])
+
+
+def _build_model():
+    w = np.array([[0.1, 0.3, 0.2, -0.1], [-0.9, 0.1, 0.5, -0.3], [0.0, -0.1, -0.4, -0.9]], dtype=np.float32)
+
+    b = np.array([0.1, 0.1, 0.1, 0.1], dtype=np.float32)
+
+    mb = ModelBuilder()
+    x = mb.add_input("X", (2, 3))
+    x = mb.add_gemm(x, w.shape, weight_data=w, bias_data=b)
+    mb.add_output(x, (2, 4))
+    return mb.build(opset_version=19, ir_version=9)
+
+
+def check_operation_count(model: onnx.ModelProto, op_type_to_count: dict[str, int]):
+    count = {}
+    for node in model.graph.node:
+        if node.op_type in op_type_to_count:
+            count[node.op_type] = count.get(node.op_type, 0) + 1
+    assert count == op_type_to_count
+
+
+def test_compress_quantize_weights_transformation():
+    model = _build_model()
+
+    x = np.array([[0.2, -0.1, 0.9], [-0.1, -0.9, 0.5]], dtype=np.float32)
+
+    # Prepare quantized model without weight compression
+    calibration_dataset = nncf.Dataset([{"X": x}])
+    quantized_model = nncf.quantize(
+        model,
+        calibration_dataset,
+        advanced_parameters=nncf.AdvancedQuantizationParameters(
+            backend_params={BackendParameters.COMPRESS_WEIGHTS: False}
+        ),
+    )
+
+    check_operation_count(quantized_model, {"QuantizeLinear": 2, "DequantizeLinear": 2})
+    compress_quantize_weights_transformation(quantized_model)
```

Comment:
This test is covering the transformation, but I would suggest to additionally test the `nncf.quantize` with `COMPRESS_WEIGHTS: True` to check that the API is working as expected.

Reply:
I don't think it is necessary here. We already have an end-to-end test (`tests/cross_fw/examples/test_examples.py[post_training_quantization_onnx_mobilenet_v2]` where we compare the model's compression rate with the reference. So we'll be able to catch the error there if `COMPRESS_WEIGHTS: True` doesn't work as expected.